packages/mdc-switch/adapter.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-unused-vars */
+
+/**
+ * Adapter for MDC Switch. Provides an interface for managing
+ * - classes
+ * - dom
+ *
+ * Additionally, provides type information for the adapter to the Closure
+ * compiler.
+ *
+ * Implement this adapter for your framework of choice to delegate updates to
+ * the component in your framework of choice. See architecture documentation
+ * for more details.
+ * https://github.com/material-components/material-components-web/blob/master/docs/code/architecture.md
+ *
+ * @record
+ */
+class MDCSwitchAdapter {
+  /** @param {string} className */
+  addClass(className) {}
+
+  /** @param {string} className */
+  removeClass(className) {}
+
+  /** @param {!EventListener} handler */
+  registerChangeHandler(handler) {}
+
+  /** @param {!EventListener} handler */
+  deregisterChangeHandler(handler) {}
+
+  /** @param {boolean} checked */
+  setChecked(checked) {}
+
+  /** @return {boolean} checked */
+  isChecked() {}
+
+  /** @param {boolean} disabled */
+  setDisabled(disabled) {}
```

Comment:
Suggestion: should we name it `setNativeDisabled()`?

it might get confused with the Foundation's `setDisabled()` method. WDYT?

Reply:
That's a good point, perhaps (although a bit longer) it should be setNativeControlChecked/setNativeControlDisabled as to match other components with native control elements.